pkg/apis/networking/v1alpha1/certificate_types.go

```diff
@@ -0,0 +1,139 @@
+
+	// Status is the current state of the Certificate.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#spec-and-status
+	// +optional
+	Status CertificateStatus `json:"status,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// CertificateList is a collection of `Certificate`.
+type CertificateList struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty"`
+
+	// Items is the list of `Certificate`.
+	Items []Certificate `json:"items"`
+}
+
+// CertificateSpec defines the desired state of Certificate.
+type CertificateSpec struct {
+	// DNSNames is a list of DNS names the Certificate could support.
+	DNSNames []string `json:"dnsNames"`
+
+	// SecretName is the name of the secret resource to store the SSL certificate in.
+	SecretName string `json:"secretName"`
+}
+
+// CertificateStatus defines the observed state of Certificate.
+type CertificateStatus struct {
+	// The detailed information related to the provisioned certificate.
+	// +optional
+	CertificateInfo CertificateInfo `json:"certificateInfo,omitempty"`
+
+	// +optional
+	Conditions duckv1alpha1.Conditions `json:"conditions,omitempty"`
+}
+
+// InitializeConditions initializes the certificate conditions.
+func (cs *CertificateStatus) InitializeConditions() {
+	certificateCondSet.Manage(cs).InitializeConditions()
+}
+
+// MarkReady marks the certificate as ready to use.
+func (cs *CertificateStatus) MarkReady() {
+	certificateCondSet.Manage(cs).MarkTrue(CertificateCondidtionReady)
+}
+
+// GetConditions returns the Conditions array. This enables generic handling of
+// conditions by implementing the duckv1alpha1.Conditions interface.
+func (cs *CertificateStatus) GetConditions() duckv1alpha1.Conditions {
+	return cs.Conditions
+}
+
+// SetConditions sets the Conditions array. This enables generic handling of
+// conditions by implementing the duckv1alpha1.Conditions interface.
+func (cs *CertificateStatus) SetConditions(conditions duckv1alpha1.Conditions) {
+	cs.Conditions = conditions
+}
+
+// GetCondition gets a speicifc condition of the Certificate status.
+func (cs *CertificateStatus) GetCondition(t duckv1alpha1.ConditionType) *duckv1alpha1.Condition {
+	return certificateCondSet.Manage(cs).GetCondition(t)
+}
+
+// ConditionType represents a Certificate condition value
+const (
+	// CertificateConditionReady is set when the requested certificate
+	// is provioned and valid.
+	CertificateCondidtionReady = duckv1alpha1.ConditionReady
+)
+
+// CertificateInfo defines the detailed information related to the provisioned
+// certificate.
+type CertificateInfo struct {
+	// The DNS Names that the certificate actually supported.
+	// SupportedDNSNames could be the super set of the DNSNames in
+	// the CertificateSpec.
```

Comment:
Do you have examples of where this would end up being a superset?  As a user, why shouldn't I see that as a bug? :)